agent/src/StreamBuffer.php

```diff
@@ -15,9 +15,9 @@ public function __construct(
         //
     }
 
-    public function write(string $input): void
+    public function write(Payload $payload): void
     {
-        $input = substr(substr($input, 1), 0, -1);
+        $input = substr($payload->value, 1, -1);
```

Comment:
Collapsed this down to a single `substr` call. No need to call it twice when you can trim the start and the end with a single call.